### ISSUE TITLE
Feature request - Flag to force overwriting output file

### ISSUE THREAD

**Issue by alexhaydock** [rabobank-cdc/DeTTECT#123]
Looking at how `write_file()` works in `file_output.py`, it seems like there's a hard-coded invocation of the `get_non_existing_filename()` function which will append a `_1` style suffix to the output file variable to be used.

https://github.com/rabobank-cdc/DeTTECT/blob/6272e9e3af14b870cda917ac3bd362a98f0c1412/file_output.py#L23

It would be useful to have a `--force-output` or `--force-overwrite` flag to control this behaviour, to make using `dettect.py` more flexible, for example for automated usage in pipelines.

If there's any interest in accepting this as an enhancement I could work on a PR.

My thinking would probably be to set a boolean variable for whether we're in `--force-overwrite` mode and pass the variable to every function in `dettect.py` where we're already passing `args.output_filename` so that we can skip `get_non_existing_filename()` with an if statement in `write_file()`.

But by my count that would involve extending the arguments accepted by the following functions:

```
export_data_source_list_to_excel
export_techniques_list_to_excel
generate_data_sources_layer
generate_detection_layer
generate_technique_administration_file
generate_visibility_layer
plot_data_sources_graph
plot_graph
```

So I wanted to type out the proposed solution to see whether it fits with the general coding practices you've adopted for the repo before trying to implement it.

**Comment by rubinatorz** [comment]
Hi @alexhaydock 

Sorry for the late response, it's a bit busy around here. But thanks for your idea and proposal. You are definitly welcome to implement this and send a PR. The direction you're pointing seems good to me.

Regards,
Ruben

**Comment by alexhaydock** [comment]
Thanks @rubinatorz!

I've made an attempt at starting this in #124. Do you have a suite of test cases to run against the repo? I tried to do some manual testing but I don't manually touch all the codepaths in DeTTECT yet so I'm not as familiar.

If I get time in the future I could look to try and contribute some test cases for regression testing but it probably wouldn't be for a little while.

**Comment by alexhaydock** [comment]
Note that I didn't address `generate_group_heat_map()` in this one since it seems to be an oddity - this function will always overwrite the output file from what I can tell:

https://github.com/rabobank-cdc/DeTTECT/blob/6272e9e3af14b870cda917ac3bd362a98f0c1412/group_mapping.py#L797-L805

I wasn't sure whether this behaviour could do with being made more consistent with the other functions as I'm not familiar enough with the codebase yet, so I left it how it is.

**Comment by rubinatorz** [comment]
Thanks for the PR. We have unittests in a private repo, so I will have a look at your #123 and run our unittests to see if everything still works as expected.

**Comment by rubinatorz** [comment]
I've looked at the PR and ran the unittests. There were some errors. At some places `overwrite_mode` did not correspond the `output_overwrite` from the function parameter. I fixed those.

In `generate_group_heat_map` it was also necessary to use the `output_overwrite` because `write_file` needs it and it also nice to have this functionality there as well. So I've added this as well to the group mode in `dettect.py`.

Thanks for the PR! I will merge it to main.
